### PR TITLE
[torch.compile] Try using Dynamo-native padding

### PR DESCRIPTION
**Pull request by ProExpertProg**
## Purpose

Reapply padding from #24666, which caused bug #25623. Currently fails with different error due to torch==2.9 (works with torch==2.8):

```
VLLM_USE_DEEP_GEMM=0 vllm serve Qwen/Qwen3-30B-A3B-FP8 --trust-remote-code
```

**UPDATE**: this now works!!

## Test Result

### CODE REVIEW

**Comment by ProExpertProg** [comment]
I tried running the full repro through lm_eval and it seems like there's an issue:

```
# serve
VLLM_USE_DEEP_GEMM=0 VLLM_ALL2ALL_BACKEND=deepep_low_latency vllm serve \
                Qwen/Qwen3-30B-A3B-FP8 \
                --port 7557 \
                --disable-uvicorn-access-log \
                --trust-remote-code \
                --enable-expert-parallel \
                --data-parallel-hybrid-lb \
                --tensor-parallel-size 2 \
                --data-parallel-size 2 \
                --data-parallel-size-local 2 \
                --data-parallel-address localhost \
                --data-parallel-rpc-port 5555 \
                --data-parallel-start-rank 0 \
                --enable-eplb \
                --eplb-config '{"window_size":"1000",
                                "step_interval":"3000",
                                "num_redundant_experts":"32",
                                "log_balancedness":"False"}' \
                --enable-dbo \
                --dbo-decode-token-threshold 32 \
                --max_num_seqs 256

# eval
lm_eval --model local-completions --tasks gsm8k --model_args model=Qwen/Qwen3-30B-A3B-FP8,base_url=http://0.0.0.0:7557/v1/completions,max_retries=3,tokenized_requests=False,timeout=1200,max_gen_toks=64,max_length=8192 --batch_size 2049 --trust_remote_code --limit 0.5
```

`main` (`949a6a19d`):
```
local-completions (model=Qwen/Qwen3-30B-A3B-FP8,base_url=http://0.0.0.0:7557/v1/completions,max_retries=3,tokenized_requests=False,timeout=1200,max_gen_toks=64,max_length=8192,trust_remote_code=True), gen_kwargs: (None), limit: 0.5, num_fewshot: None, batch_size: 2049
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.1955|±  |0.0154|
|     |       |strict-match    |     5|exact_match|↑  |0.1091|±  |0.0121|
```

This PR:
```
local-completions (model=Qwen/Qwen3-30B-A3B-FP8,base_url=http://0.0.0.0:7557/v1/completions,max_retries=3,tokenized_requests=False,timeout=1200,max_gen_toks=64,max_length=8192,trust_remote_code=True), gen_kwargs: (None), limit: 0.5, num_fewshot: None, batch_size: 2049
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.0061|±  | 0.003|
|     |       |strict-match    |     5|exact_match|↑  |0.0000|±  | 0.000|
```

File: vllm/model_executor/layers/quantization/utils/fp8_utils.py [hunk 1]
```diff
@@ -139,6 +139,33 @@ def _padded_cutlass(
...
+    # TODO(torch==2.10): torch semantics broken for PythonMod in 2.9:
+    #  so x<=0 -> x % y <= 0 (not actually python mod semantics).
+    #  https://github.com/pytorch/pytorch/issues/169602
+    #  We increment the padded dim to avoid a negative.
+    # padding = -qx.shape[0] % pad_multiple
+    padding = (qx.shape[0] * pad_multiple - qx.shape[0]) % pad_multiple
+    # print(f"{padding=}")
+    padded_qx = torch.nn.functional.pad(qx, (0, 0, 0, padding))
+    assert len(x_scale.size()) == 2
+    padded_x_scale = torch.nn.functional.pad(x_scale, (0, 0, 0, padding), value=1.0)
```
Comment (ElizaWszola) [inline vllm/model_executor/layers/quantization/utils/fp8_utils.py:161]: Looks like `torch.nn.functional.pad` doesn't preserve the col-major layout of scales and the returned tensor is row-major - this leads to bad output of `cutlass_scaled_mm`

Comment (ProExpertProg) [reply vllm/model_executor/layers/quantization/utils/fp8_utils.py:161]: Nice find!

Comment (ElizaWszola) [reply vllm/model_executor/layers/quantization/utils/fp8_utils.py:161]: I checked the transpose trick and it works - can you verify?

```suggestion
    padded_x_scale = torch.nn.functional.pad(
        x_scale.transpose(0, 1), (0, padding, 0, 0), value=1.0
    ).transpose(0, 1)
```
